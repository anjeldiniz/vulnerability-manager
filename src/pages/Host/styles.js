import styled from "styled-components";

export const Container = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
  gap: 10px;
  height: 100%;
`;

export const Title = styled.h2`
margin-top: 20px;
`;

export const Table = styled.table`
  color: #000318;
  font-size: 100%;
  height: 100%;
  width: 100%;
  border: 10px solid #ddd;
  border-radius: 5px;
  padding: 5px 5px;
  text-align: center;
  margin-top: 20px;
`;

export const Th =  styled.th `
  background-color: #1A202C;
  color: white;
  border: 1px solid #1A202C;
  border-radius: 5px;
  border-collapse: collapse;
  padding: 4px;
`;

export const Td =  styled.td `
  border: 1px solid #ddd;
  border-radius: 5px;
  border-collapse: collapse;
  padding: 4px;
  height: 70px;
  &:hover {
    background-color: #046ee5;
    color: white;
  }

`;

export const Pagination = styled.div`
  display: flex;
  min-width: 500px;
  max-width: 900px;
  justify-content: space-between;
  margin-top: 10px;
`;

export const PaginationButton = styled.div`
  display: flex;
`;

export const PaginationItem = styled.div`
  margin: 0 10px;
  cursor: pointer;

  ${(props) =>
    props.isSelect && {
      background: "#6d6d6d",
      padding: "0 5px",
    }}
`;

