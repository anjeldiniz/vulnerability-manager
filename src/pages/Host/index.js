import React, { useEffect } from 'react';
import Header from "../../components/Header/index";
import moment from "moment";
import useAuth from '../../hooks/useAuth';
import * as C from './styles';


function Host() {
  const { results, getAssets } = useAuth();

  useEffect(() => {
    getAssets()
  },);

  return (
    <>
    <Header />
    <C.Container>
      <C.Title>About Volun Count</C.Title>
      <C.Container>
        <C.Table>
          <thead>
            <tr>
              <C.Th>Name</C.Th>
              <C.Th>Risk</C.Th>
              <C.Th>Volun Count</C.Th>
              <C.Th>IP Address</C.Th>
              <C.Th>Created</C.Th>
              <C.Th>Modified</C.Th>
            </tr>
          </thead>
          <tbody>
            {
            results
            ?.sort((a, b) => b.vuln_count - a.vuln_count)
            ?.map((el, index) => (
              <tr className='row' key={index}>
                <C.Td>{el.hostname}</C.Td>
                <C.Td>{el.risk}</C.Td>
                <C.Td>{el.vuln_count}</C.Td>
                <C.Td>{el.ip_address}</C.Td>
                <C.Td>{moment(el.created).format("DD/MM/YYYY")}</C.Td>
                <C.Td>{moment(el.modified).format("DD/MM/YYYY")}</C.Td>
              </tr>
            ))}
          </tbody>
        </C.Table>
      </C.Container>
    </C.Container>
    </>
  );
}

export default Host;