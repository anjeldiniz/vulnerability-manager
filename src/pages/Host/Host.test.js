import { render } from '@testing-library/react';
import { BrowserRouter as Router } from "react-router-dom";
import Host from "./";
import AuthProvider from "../../contexts/authProvider";

describe("Test the Host component", () => {
  it("Should match the snapshot", () => {
    const { asFragment } = render(<AuthProvider><Router><Host /></Router></AuthProvider>)
    expect(asFragment()).toMatchSnapshot();
  })
})
