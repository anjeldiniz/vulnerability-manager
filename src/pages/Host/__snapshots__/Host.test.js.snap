// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the Host component Should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-bcXHqe llzPbo"
  >
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 448 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
      />
    </svg>
    <svg
      class="logout"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 512 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm0 448c-110.5 0-200-89.5-200-200S145.5 56 256 56s200 89.5 200 200-89.5 200-200 200zm101.8-262.2L295.6 256l62.2 62.2c4.7 4.7 4.7 12.3 0 17l-22.6 22.6c-4.7 4.7-12.3 4.7-17 0L256 295.6l-62.2 62.2c-4.7 4.7-12.3 4.7-17 0l-22.6-22.6c-4.7-4.7-4.7-12.3 0-17l62.2-62.2-62.2-62.2c-4.7-4.7-4.7-12.3 0-17l22.6-22.6c4.7-4.7 12.3-4.7 17 0l62.2 62.2 62.2-62.2c4.7-4.7 12.3-4.7 17 0l22.6 22.6c4.7 4.7 4.7 12.3 0 17z"
      />
    </svg>
  </div>
  <div
    class="sc-eDvSVe eHYKbL"
  >
    <h2
      class="sc-jSUZER drupwc"
    >
      About Volun Count
    </h2>
    <div
      class="sc-eDvSVe eHYKbL"
    >
      <table
        class="sc-gKPRtg fvPwYa"
      >
        <thead>
          <tr>
            <th
              class="sc-iBYQkv hDGMru"
            >
              Name
            </th>
            <th
              class="sc-iBYQkv hDGMru"
            >
              Risk
            </th>
            <th
              class="sc-iBYQkv hDGMru"
            >
              Volun Count
            </th>
            <th
              class="sc-iBYQkv hDGMru"
            >
              IP Address
            </th>
            <th
              class="sc-iBYQkv hDGMru"
            >
              Created
            </th>
            <th
              class="sc-iBYQkv hDGMru"
            >
              Modified
            </th>
          </tr>
        </thead>
        <tbody />
      </table>
    </div>
  </div>
</DocumentFragment>
`;
